processors/db-processor/src/main/kotlin/net/corda/processors/db/internal/reconcile/db/DbReconcilerReader.kt

```
@@ -96,19 +97,24 @@ class DbReconcilerReader<K : Any, V : Any>(
     @Suppress("SpreadOperator")
     override fun getAllVersionedRecords(): Stream<VersionedRecord<K, V>>? {
         return try {
-            val streams = reconciliationContextFactory.invoke().map { context ->
-                val currentTransaction = context.entityManager.transaction
-                currentTransaction.begin()
-                doGetAllVersionedRecords(context).onClose {
-                    // This class only have access to this em and transaction. This is a read only transaction,
-                    // only used for making streaming DB data possible.
-                    currentTransaction.rollback()
-                    context.close()
+            reconciliationContextFactory().mapNotNull { context ->
+                try {
+                    val currentTransaction = context.getOrCreateEntityManager().transaction
+                    currentTransaction.begin()
+                    doGetAllVersionedRecords(context).onClose {
+                        // This class only have access to this em and transaction. This is a read only transaction,
+                        // only used for making streaming DB data possible.
+                        currentTransaction.rollback()
+                        context.close()
+                    }
+                } catch (e: Exception) {
+                    logger.warn("Error while processing reconciliation record stream", e)
+                    coordinator.postEvent(GetRecordsErrorEvent(e))
+                    null
```

I think returning `null` here means we will be swallowing the issue and proceeding with rest of queries. In case of vnodes would mean that DB records fetched for one vnode failed and we just continue with rest vnodes. Sounds like we would want to `throw e` instead of `null`, which will get thrown when we `use` the stream i.e. in `ReconcilerEventHandler.reconcile()` which gets handled by caller of `reconcile()`

---

Sure I can change that as long as the exception is caught elsewhere

---

Yes they are at `ReconcilerEventHandler`. So I think we should still notify `DbReconcilerReader` with posting that event but then throw instead of `null`. `null` was there in the first place to notify the caller of this method synchronously that couldn't retrieve DB results, because `ReconcilerEventHandler` would  get notified asynchronously via domino logic.

---

So we should keep existing try catch to have the chance to notify `DbReconcilerReader` only change `null` to rethrow exception I think.